File: packages/mobile/ios/App/App/OpenChamberPhysicalScalePlugin.swift
import Capacitor
import UIKit

@objc(OpenChamberPhysicalScalePlugin)
class OpenChamberPhysicalScalePlugin: CAPPlugin, CAPBridgedPlugin {
    let identifier = "OpenChamberPhysicalScalePlugin"
    let jsName = "OpenChamberPhysicalScale"
    let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getMetrics", returnType: CAPPluginReturnPromise),
    ]

    @objc func getMetrics(_ call: CAPPluginCall) {
        // 1 CSS px = 1 UIKit pt ≈ 1/163 in. Web scale stays 1.
        call.resolve([
            "xdpi": 163.0,
            "ydpi": 163.0,
            "density": 1.0,
            "densityDpi": 163,
            "stableDensityDpi": 163,
        ])
    }
}

File: packages/mobile/android/app/src/main/java/com/openchamber/app/OpenChamberPhysicalScalePlugin.java
package com.openchamber.app;

import android.util.DisplayMetrics;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "OpenChamberPhysicalScale")
public class OpenChamberPhysicalScalePlugin extends Plugin {
    @PluginMethod
    public void getMetrics(PluginCall call) {
        DisplayMetrics metrics = getContext().getResources().getDisplayMetrics();
        JSObject result = new JSObject();
        result.put("xdpi", (double) metrics.xdpi);
        result.put("ydpi", (double) metrics.ydpi);
        result.put("density", (double) metrics.density);
        result.put("densityDpi", metrics.densityDpi);
        result.put("stableDensityDpi", DisplayMetrics.DENSITY_DEVICE_STABLE);
        call.resolve(result);
    }
}
